src/notebooks/controllers/notebookControllerManager.ts

```diff
@@ -472,13 +492,46 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
             let preferredConnection: KernelConnectionMetadata | undefined;
             // Don't attempt preferred kernel search for interactive window, but do make sure we
             // load all our controllers for interactive window
+            const notebookMetadata = getNotebookMetadata(document);
             if (document.notebookType === JupyterNotebookView) {
-                preferredConnection = await this.kernelFinder.findKernel(
-                    document.uri,
-                    getNotebookMetadata(document),
-                    preferredSearchToken.token
+                const resourceType = getResourceType(document.uri);
+                const isPythonNbOrInteractiveWindow =
+                    isPythonNotebook(notebookMetadata) || resourceType === 'interactive';
+                const preferredInterpreter =
+                    isPythonNbOrInteractiveWindow && this.extensionChecker.isPythonExtensionInstalled
+                        ? await this.interpreters.getActiveInterpreter(document.uri)
+                        : undefined;
+
+                ({ preferredConnection } = await this.findPreferredKernelExactMatch(
+                    document,
+                    notebookMetadata,
+                    preferredSearchToken.token,
+                    'useCache',
+                    preferredInterpreter
+                ));
+
+                // If we don't get a preferred, prep te ignore cache search to check next
+                const ignoreCacheFindPreferredPromise = this.findPreferredKernelExactMatch(
```

Comment:
shouldn't this call go inside the if statement? or is the point to refresh the cache either way?

Reply:
Yeah, Rich had a similar comment. This mimics how the other call to list kernels in the controller already works, and mimics how the old code used to work. But that doesn't mean that it's right ;). I'm looking into it now. I believe that the intent was that the cache should be refreshed each time, trying to confirm that.